internal/mode/static/state/dataplane/configuration.go

```diff
@@ -637,6 +634,26 @@ func buildTelemetry(g *graph.Graph) Telemetry {
 		tel.Interval = string(*telemetry.Exporter.Interval)
 	}
 
+	// FIXME(sberman): https://github.com/nginxinc/nginx-gateway-fabric/issues/2038
+	// Find a generic way to include relevant policy info at the http context so we don't need policy-specific
+	// logic in this function
+	ratioMap := make(map[string]int32)
+	for _, pol := range g.NGFPolicies {
+		if obsPol, ok := pol.Source.(*ngfAPI.ObservabilityPolicy); ok {
+			if obsPol.Spec.Tracing != nil && obsPol.Spec.Tracing.Ratio != nil && *obsPol.Spec.Tracing.Ratio > 0 {
+				ratioName := observability.CreateRatioVarName(obsPol)
+				ratioMap[ratioName] = *obsPol.Spec.Tracing.Ratio
+			}
+		}
+	}
+
+	if len(ratioMap) > 0 {
```

Comment:
is this `if` redundant?

Reply:
Yeah, though the main thing behavior it change is whether or not the `tel.Ratios` list is nil or initialized to a 0 length slice. Don't know if it really matters either way.

Reply:
> Yeah, though the main behavior it changes is whether or not the tel.Ratios list is nil or initialized to a 0 length slice.

our client code doesn't depend on slice being nil or not. 
so often we deliberately use nil so that it is easier to fill expected structures in unit tests

Reply:
I removed this check and updated the tests accordingly, think I should put it back?

Reply:
I think it looks good 👍